### ISSUE TITLE
running dots hard to see

### ISSUE THREAD

**Issue by axaxs** [micheleg/dash-to-dock#170]
Hi,

Very nice extension, and running dots is a great addition.  However, when any of the following are true, the dots are very hard to see - 
1 - many items open
2 - icon size set to smallest value
3 - opacity set to about any translucency(esp around 75%)

Is this by design, or could some better contrasting color be used for the dots?  I've been manually changing them in an svg editor, but thought others might have this issue too so opening here.

**Comment by micheleg** [comment]
Not a design decision. I tried to reuse the color of the highlights in the default theme - i.e. light gray in 3.14 and light blue in 3.16. As you noticed, the colour can easily be changed in the svgs files, and a theme author can support the extension by providing their files. Do you think the dots are difficult to see with the default theme? What colour would you suggest?

**Comment by axaxs** [comment]
On a high res 22" lcd monitor they work well enough.  On my 14 inch chromebook, not so much.  I played around a bit, and the easiest to see color are shades of orange.  Though, that may look tacky.  I'd actually suggest on a dual tone, maybe a light grey circle with a dark grey/black ring around it.  Shades of blue and gray get lost in the panel color, and lighter colors get lost in typically white backgrounds when transparency is set.

**Comment by micheleg** [comment]
The way I think those dots, they don't have to stand up and scream at you constantly, but just quickly inform you about the number of opened windows when needed. I did try using orange as a 'notification' flag some time ago and that is more visible, but it completely out of place in the default theme which is what I'm mainly targeting. I looked into a way to change that colour from settings, but unless I replace the background svgs with proper elements which are css themable I don't think it's possible, or I didn't manage to find the way of doing it.

In my experience I never have visibility problems but that is because I use the dock with an almost opaque black background. I like your suggestion of the two-colour dots as they would improve the contrast in all conditions and I'll put this option in my todo list. If you make for yourself some svg background that you find working well, please consider sending them to me so that I can test them.

**Comment by user501254** [comment]
I modified the existing SVGs to add a white stroke around the circles. It Doesn't does much.

| Orignal | on Default Adwaita theme | on Arc theme with Paper icons |
| ------------- | ------------- | ------------- |
| ![screenshot from 2015-10-04 19-03-15](https://cloud.githubusercontent.com/assets/6129517/10268068/c4dc5bc8-6aca-11e5-814d-c821a741a538.png) | ![screenshot from 2015-10-04 18-39-47](https://cloud.githubusercontent.com/assets/6129517/10268009/b3859454-6ac8-11e5-93b9-d898e05898a4.png)  | ![screenshot from 2015-10-04 18-38-29](https://cloud.githubusercontent.com/assets/6129517/10268012/cbee7bdc-6ac8-11e5-8abe-0078916ed914.png)  |

**Comment by micheleg** [comment]
H, would you mind uploading the svgs so that I can test them?

**Comment by user501254** [comment]
Sure, how would you like me to upload them? A pull-request or over here as a compressed file?

**Comment by micheleg** [comment]
As you prefer, a pull request would be easier for me.

**Comment by micheleg** [comment]
I haven't thought too much about the new style, mainly because I still use Gnome Shell 3.14 where the dots are light grey and very well visible. However, I spent some time reworking the way dots are shown. I now generate them programmatically instead of using background svgs. Therefore, the style can now be defined by css. I'm reusing the style of the default "dot", so it should integrate better with custom themes. A border can be set by styling the class 'app-well-app-running-dot'.

These changes are in the branch rework_running_indicators [https://github.com/micheleg/dash-to-dock/tree/rework_running_indicators]

**Comment by user501254** [comment]
Since you are going for CSS would it be feasible to add a color picker within the settings menu so that users can pick their own desired style?

**Comment by micheleg** [comment]
I updated the branch with settings to choose the dot color, border-color and border-width.

**Comment by user501254** [comment]
Just tried out the updated branch, Good job!
The ability to choose custom colors has solved one more issue that I had while using some third party themes, see: https://github.com/horst3180/Arc-theme/issues/263
![screenshot from 2015-11-01 13-08-57](https://cloud.githubusercontent.com/assets/6129517/10867763/4b9d6924-809a-11e5-989a-35cfcdf7c55d.png)

**Comment by lpinner** [comment]
I just tried the rework_running_indicators branch, works well, much easier to see the dots. I had a lot of trouble seeing the original dots using the [paper theme](http://samuelhewitt.com/paper).

**Comment by micheleg** [comment]
Good. This has no been merged into master [a76837d, 7afa750, a82c45213d,  afc7f4f304, 8af0dfa8b2, 5862c087c9], and backported to 3.16 too [6c2f6c423a, d05e560f74, 2ad6f20, aaf1b69e, b7238d87, ecec82251f ].

**Comment by zagortenay333** [comment]
Wait, so theme developers can no longer style the dots?

**Comment by micheleg** [comment]
On the contrary: I believe now it is even easier for theme writer to customize the dock:
1. the extension by default inherit the style of the "running dot" - the default blue rectangle -  which means that if you want circles as indicators you just have to specify its colour.
2. If you were applying more complex styles, exactly as before you can hide the default dot and play with backgrounds or whatever you want. 

Basically, it 's better for themes which are not specifically targeting dash-to-dock, as I try to inherit the style and allow the user to correct for it. Nothing change for themes targeting dash-to-dock. If you noticed some regression, and it doesn't work like this, than there is a bug.

**Comment by zagortenay333** [comment]
Yeah, but I can't make custom indicators for multiple running instances?
![screenshot from 2015-12-01 00-05-32](https://cloud.githubusercontent.com/assets/11144630/11487414/46cd970c-97bf-11e5-9b81-38b6b608652b.png)

**Comment by micheleg** [comment]
I didn't intentionally break anything: can you give me an example of what worked before and doesn't work now. I don't understand your screenshot.

**Comment by zagortenay333** [comment]
When I use a custom indicator, I can't create custom images for running1, running2... Notice how they use different color.

**Comment by micheleg** [comment]
My fault, I did indeed remove the classes. I could re-enable them, you are right.

**Comment by zagortenay333** [comment]
Thx that would be cool.
Btw, would it be possible for the image to be positioned on the side that is towards the edge of the screen?
The only way to acomplish that now is by making the image be 96x96, but then it must be scaled down for smaller icon sizes.
Basically, I'd like if those indicators bars could be on the left side of the icons, not the bottom.

**Comment by micheleg** [comment]
There is no real technical reason for not applying those classes, beside the fact I don't use them any more. But I agree they are very handy for custom theme development which I care to support as much as possible. So I will bring them back.

Regarding your question, when I looked into it I end up making large square images as well. For compatibility with the default dash styling I didn't change on purpose the layout of the icon containers. I don't know if there is some clever css rule supported by the shell that can help you. Which is the styling you can't achieve with large background images? Even if they were placed on the side you would need to scale them, or am I misunderstanding you?

**Comment by zagortenay333** [comment]
Well, I would use the `background-image` property with a fixed width/height (say 24x24), and it would be cool if it could be oriented the same way the dots are. So towards the edge, but if it can't be done np.

**Comment by micheleg** [comment]
I looked for a way to rotate the background, but I didn't find one. I used to generate all possible rotations of the background images, and then apply them according to the .left .right .bottom .top class, which is applied to the #dashtodockContainer element. Tedious but was working for me. Have a look at my old internal stylesheet:
https://github.com/micheleg/dash-to-dock/blob/cbcc067c3d221ac8b319993bbcb8266bf74212a8/stylesheet.css

**Comment by zagortenay333** [comment]
The problem with that is that I end up with blurry images when they get scaled.

**Comment by micheleg** [comment]
Even if you use svgs?

**Comment by zagortenay333** [comment]
Yeah ofc. Svg don't help much with small icons and downwards scaling.

**Comment by micheleg** [comment]
Do you have a link for your theme?

**Comment by zagortenay333** [comment]
https://github.com/zagortenay333/ciliora-tertia-shell

**Comment by micheleg** [comment]
I'll think of it. So you can style correctly (without blur) the runningdot element which is on the bottom of the icon. And you would like it to be placed and oriented according to the dock position on the external side of the screen. That would be enough? I can't promise anything though.

**Comment by zagortenay333** [comment]
Thx! Oh, and thx for helping out theme developers! :smile: Wish more extension devs did that.

**Comment by micheleg** [comment]
```running1, ..., running4``` css classes have been readded [ fc05e78d0dcdb65e15ba30f1417b9bde599a04e0].
